.azure/pipelines/ci.yml

```diff
@@ -780,12 +772,72 @@ stages:
             - Linux_musl_arm_build
             - Linux_musl_arm64_build
-          # In addition to the dependencies above, ensure the build was successful overall.
+          # In addition to the dependencies above that provide assets, ensure the build was successful overall.
+          - ${{ if in(variables['Build.Reason'], 'Manual') }}:
+            - Code_check
+            - Windows_Test
+            - MacOS_Test
+            - Linux_Test
+            - Helix_x64
           - Source_Build_Managed
         pool:
           vmImage: vs2017-win2016
         publishUsingPipelines: ${{ variables._PublishUsingPipelines }}
         enablePublishBuildArtifacts: true # publish artifacts/log files
+  - ${{ if and(ne(variables['System.TeamProject'], 'public'), eq(variables['Build.SourceBranch'],'refs/heads/main')) }}:
+    - template: /eng/common/templates/job/source-index-stage1.yml
+      parameters:
+        binlogPath: $(System.ArtifactsDirectory)/Windows_arm_Logs/Release/Build.binlog
+        dependsOn:
+          - Windows_arm_build
+          # In addition to the dependency above that provides assets, ensure the build was successful overall.
+          - Windows_build
+          - Windows_arm64_build
+          - ${{ if ne(variables['PostBuildSign'], 'true') }}:
+            - CodeSign_Xplat_MacOS_arm64
+            - CodeSign_Xplat_MacOS_x64
+            - CodeSign_Xplat_Linux_x64
+            - CodeSign_Xplat_Linux_arm
+            - CodeSign_Xplat_Linux_arm64
+            - CodeSign_Xplat_Linux_musl_x64
+            - CodeSign_Xplat_Linux_musl_arm
+            - CodeSign_Xplat_Linux_musl_arm64
+          - ${{ if eq(variables['PostBuildSign'], 'true') }}:
+            - MacOs_arm64_build
+            - MacOs_x64_build
+            - Linux_x64_build
+            - Linux_arm_build
+            - Linux_arm64_build
+            - Linux_musl_x64_build
+            - Linux_musl_arm_build
+            - Linux_musl_arm64_build
+          - ${{ if in(variables['Build.Reason'], 'Manual') }}:
+            - Code_check
+            - Windows_Test
+            - MacOS_Test
+            - Linux_Test
+            - Helix_x64
+          - Source_Build_Managed
+        pool:
+          name: NetCore1ESPool-Internal
+          # Visual Studio Enterprise - no BuildTools agents exist internally and job must run on Windows
+          demands: ImageOverride -equals Build.Server.Amd64.VS2019
+        preSteps:
+          - task: DownloadBuildArtifacts@0
+            inputs:
+              artifactName: Windows_arm_Logs
+              cleanDestinationFolder: true
+              itemPattern: '**/Build.binlog'
+          - task: DownloadBuildArtifacts@0
+            inputs:
+              artifactName: Windows_arm_csharp
+              cleanDestinationFolder: true
+          - task: CopyFiles@2
+            inputs:
+              sourceFolder: '$(System.ArtifactsDirectory)/Windows_arm_csharp/'
+              targetFolder: '$(Build.SourcesDirectory)'
```

Comment:
Can't download the sources directly into `$(Build.SourcesDirectory)` because `DownloadBuildArtifacts@0` unconditionally puts everything into a Windows_arm_csharp/ folder. That means the files are (again) not where `BinLogToSln` expects them.